### PR TITLE
Combo Boluses initial release

### PR DESCRIPTION
**Pull request by MilosKozak**

### CODE REVIEW

File: lib/client/careportal.js [hunk 1]
```diff
@@ -121,9 +126,26 @@ function init (client, $) {
...
+      $('#insulinSplitExt').val(100 - nowval);
+      $('#insulinSplitNow').val(nowval);
+    } else {
+      var extval = parseInt($('#insulinSplitExt').val()) || 0;
+      $('#insulinSplitNow').val(100 - extval);
+      $('#insulinSplitExt').val(extval);
+    }
+  
+    maybePrevent(event);
+  }
```
Comment (jasoncalabrese) [inline lib/client/careportal.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1478533780/issues/source?bid=2452379&fileBranchId=2998783#l147)

File: lib/profilefunctions.js [hunk 2]
```diff
@@ -216,18 +229,27 @@ function init(profileData) {
 
     var basal = profile.getBasal(time, spec_profile);
     var tempbasal = basal;
+    var combobolusbasal = 0;
+    var totalbasal = basal;
```
Comment (jasoncalabrese) [inline lib/profilefunctions.js]: Codacy Issue found: ['totalbasal' is defined but never used.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1498320447/issues/source?bid=2452379&fileBranchId=2998783#l233)

File: static/report/js/report.js [hunk 3]
```diff
@@ -448,7 +448,13 @@
     options.maxInsulinValue = maxInsulinValue;
     options.maxCarbsValue = maxCarbsValue;
 
-
+    datastorage.treatments = [];
+    datastorage.combobolusTreatments = [];
+    for (var day in daystoshow) {
```
Comment (jasoncalabrese) [inline static/report/js/report.js]: Codacy Issue found: [The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1498320474/issues/source?bid=2452379&fileBranchId=2998783#l453)

File: lib/plugins/basalprofile.js [hunk 4]
```diff
@@ -57,8 +62,18 @@ function init() {
       info.push({label: 'Active temp basal remaining', value: remaining + ' min'});
     }
       
+    if (basalValue.combobolustreatment) {
+      tempText = (basalValue.combobolustreatment.relative ? '+' + basalValue.combobolustreatment.relative + 'U/h' : '');
```
Comment (jasoncalabrese) [inline lib/plugins/basalprofile.js]: Codacy Issue found: ['tempText' used out of scope.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1498320423/issues/source?bid=2452379&fileBranchId=2998783#l66)

File: lib/plugins/basalprofile.js [hunk 5]
```diff
@@ -57,8 +62,18 @@ function init() {
...
+      remaining = parseInt(basalValue.combobolustreatment.duration - times.msecs(sbx.time - basalValue.combobolustreatment.mills).mins);
+      info.push({label: '------------', value: ''});
+      info.push({label: 'Active combo bolus', value: tempText});
```
Comment (jasoncalabrese) [inline lib/plugins/basalprofile.js]: Codacy Issue found: ['tempText' used out of scope.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1498320423/issues/source?bid=2452379&fileBranchId=2998783#l69)

File: lib/plugins/basalprofile.js [hunk 6]
```diff
@@ -57,8 +62,18 @@ function init() {
...
+      info.push({label: 'Active combo bolus start', value: new Date(basalValue.combobolustreatment.mills).toLocaleString()});
+      info.push({label: 'Active combo bolus duration', value: parseInt(basalValue.combobolustreatment.duration) + ' min'});
+      info.push({label: 'Active combo bolus remaining', value: remaining + ' min'});
```
Comment (jasoncalabrese) [inline lib/plugins/basalprofile.js]: Codacy Issue found: ['remaining' used out of scope.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1498320423/issues/source?bid=2452379&fileBranchId=2998783#l72)

File: lib/plugins/basalprofile.js [hunk 7]
```diff
@@ -57,8 +62,18 @@ function init() {
       info.push({label: 'Active temp basal remaining', value: remaining + ' min'});
     }
       
+    if (basalValue.combobolustreatment) {
+      tempText = (basalValue.combobolustreatment.relative ? '+' + basalValue.combobolustreatment.relative + 'U/h' : '');
+      remaining = parseInt(basalValue.combobolustreatment.duration - times.msecs(sbx.time - basalValue.combobolustreatment.mills).mins);
```
Comment (jasoncalabrese) [inline lib/plugins/basalprofile.js]: Codacy Issue found: ['remaining' used out of scope.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1498320423/issues/source?bid=2452379&fileBranchId=2998783#l67)

**Comment by mgranberry** [comment]
Thanks.  This is working great with my t:slim uploads.

**Comment by jasoncalabrese** [comment]
Looks good, was nice to see there weren't any conflicts with my devicestatus branch, going to merge it

**Comment by bewest** [comment]
Great :-)

**Comment by mgranberry** [comment]
It seems a little awkward in this particular event where I massively overcorrected, but scheduled a combo bolus to take care of the carbs before I was actually completely out of the woods.  It's going to wipe out my basal display for a while.

untitled

**Comment by mgranberry** [comment]
I also just noticed that there is no display for the 42g of carbs that I entered along with the start of the combo bolus.

**Pull request merged**